File: src/IMG/IMG_III.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package IMG;

import file_io.Message;
import file_io.ReadFunctions;
import shadowmapper.Util;
import java.util.ArrayList;

/**
 *
 * @author Kilian
 */
public class IMG_III {

    public void loadImg(IMG image) {
        ArrayList<IMG_Item> items = new ArrayList();

        ReadFunctions rf = new ReadFunctions();
        rf.openFile(image.getFileName().replace(".img", ".dir"));

        int itemCount = 0;

        while(rf.moreToRead() != 0){
            IMG_Item item = new IMG_Item();
            int itemOffset = rf.readInt()*2048;
            int itemSize = rf.readInt()*2048;
            String itemName = rf.readNullTerminatedString(24);
            int itemType = Util.getFileType(itemName, image);
            Message.displayMsgHigh("Offset: " + Util.getHexString(itemOffset));
            Message.displayMsgHigh("Size: " + itemSize + " bytes");
            Message.displayMsgHigh("Name: " + itemName);
            Message.displayMsgHigh("Type: " + itemType);
            item.setOffset(itemOffset);
            item.setName(itemName);
            item.setSize(itemSize);
            item.setType(itemType);
            items.add(item);
            itemCount++;
        }

        Message.displayMsgHigh("Final Item Count: " + itemCount);

        if(items.size() > 0) image.setItems(items);
        else image.setItems(null);
    }

}

File: src/main/java/nl/shadowlink/tools/shadowlib/img/IMG_III.kt
package nl.shadowlink.tools.shadowlib.img

import nl.shadowlink.tools.io.ReadFunctions
import nl.shadowlink.tools.shadowlib.utils.Utils

/**
 * @author Shadow-Link
 */
class IMG_III {

    fun loadImg(image: IMG) {
        val items = ArrayList<IMG_Item>()

        val rf = ReadFunctions(image.fileName.replace(".img", ".dir"))

        var itemCount = 0

        while (rf.moreToRead() != 0) {
            val item = IMG_Item()
            val itemOffset = rf.readInt() * 2048
            val itemSize = rf.readInt() * 2048
            val itemName = rf.readNullTerminatedString(24)
            val itemType = Utils.getFileType(itemName)
            // Message.displayMsgHigh("Offset: " + Utils.getHexString(itemOffset));
            // Message.displayMsgHigh("Size: " + itemSize + " bytes");
            // Message.displayMsgHigh("Name: " + itemName);
            // Message.displayMsgHigh("Type: " + itemType);
            item.offset = itemOffset
            item.name = itemName
            item.size = itemSize
            item.type = itemType
            items.add(item)
            itemCount++
        }

        image.items = items
        // Message.displayMsgHigh("Final Item Count: " + itemCount);
    }
}
